app/services/ems_infra_dashboard_service.rb

```diff
@@ -125,12 +125,13 @@ def ems_utilization
   end
 
   def daily_provider_metrics
-    current_user = @controller.current_user
-    tp = TimeProfile.profile_for_user_tz(current_user.id, current_user.get_timezone) || TimeProfile.default_time_profile
-
-    @daily_metrics ||= Metric::Helper.find_for_interval_name('daily', tp)
```

Comment:
`find_for_interval_name` specifically has logic to include time profiles from other regions because it assumes that metrics will be aggregated across multiple regions when we're calling from the global.

I'm wondering if the change to using a single time profile will break this for the case when were on the global region and want the metrics for the providers from all the remote regions. I only bring it up because if this was only used for a specific ems (and therefore didn't need to care about aggregating metrics which may have timeprofiles from different regions) then we wouldn't have the `|| ManageIQ::Providers::InfraManager.all` fallback.

So TLDR is: Does this method get used in cases where we want to aggregate metrics across all providers in the global region?

Reply:
I don't know, who can help us ? I also don't know why we need to include metric rollups with all time profiles. cc @kbrock ?

Reply:
@gtanzillo maybe you can weigh in here? I think maybe also a UI team member to tell us when we would use this method when `@ems` is not set.

Reply:
this is original PR https://github.com/ManageIQ/manageiq/pull/11643

> I think maybe also a UI team member to tell us when we would use this method when @ems is not set.
^^ @h-kataria

Reply:
I think the issue is that when there are providers that live in the global region daily rollups are generating for every time profile across all regions.

The capture would be indirectly scoped to the region because it's scoped to the MiqServer's zone. See [here](https://github.com/ManageIQ/manageiq/blob/0bbfe674f814f104d9a8db4673267bc6a7d257be/app/models/miq_schedule_worker/jobs.rb#L71). So that should be ok.

But the rollup to daily that gets called during the processing does not scope the time profiles to the current region. See [here](https://github.com/ManageIQ/manageiq/blob/5498df750b797e6c599a7afbe5a0b2c024adfcce/app/models/metric/ci_mixin/rollup.rb#L26) and [here](https://github.com/ManageIQ/manageiq/blob/b6bd309650ee46fedcdf6d9e00af4f06685bd7f8/app/models/metric/helper.rb#L101). The second method gets profiles from all regions. I think that method needs to scope to the current region to ensure that we only roll up metrics for the current region for only time profiles in the current region.

What do you think @lpichler @carbonin

Reply:
Right @gtanzillo that was the original issue (second bullet in https://github.com/ManageIQ/manageiq-ui-classic/pull/6544#issuecomment-565520073).

My concern here is that the page looks like it was created to handle a multiple ems case. And moving away from `find_for_interval_name` prevents this method from finding metrics from any other region (which would be associated with a remote time profile).

My question is still "Is this method ever used for aggregating metrics across regions?"
@h-kataria?

Reply:
@carbonin `@ems`  here is referring to the provider record that's being viewed in UI, but i am not sure if this is supposed to aggregate metrics across regions.

Reply:
If `@ems` is always set and we are always looking at just a single provider then we can't possibly be looking at multiple regions. If that's the case @h-kataria then this should be safe to merge.

Reply:
@carbonin yes `@ems` is always set in this case.